### PR TITLE
Fix #17298 - Analize bb with branch delayed instruction head ##anal

### LINKED ISSUE #17298: sparc missing basic block in cfg
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Arch Linux Intel 64 bit
| File format of the file you reverse (mandatory)      | ELF
| Architecture/bits of the file (mandatory)            | Sparc
| r2 -v full output, **not truncated** (mandatory)     | radare2 4.5.0-git 25153 @ linux-x86-64 git.4.4.0-429-ga933ba8be commit: a933ba8bebab7c97b8ffdb56ee8bb5394cfbab2e build: 2020-07-16__17:18:05

### Expected behavior
The `ld [i5+0x3e8], o0` instruction should not cause the end of a basic block unless another basic block is pointing to the next instruction.

### Actual behavior
Running `af` on a function produces a basic block containing only `ld [i5+0x3e8], o0`, the next instruction (+4 bytes) is not considered a part of the function and is not a part of any basic blocks. The next basic block starts at +96 bytes. This results in code being hidden in the `VV` view.

### Steps to reproduce the behavior 
```
$ r2 elf/elf-solaris-sparc-ls 
 -- Welcome to "IDA - the roguelike"
[0x00012d28]> s 0x0001bc04
[0x0001bc04]> af
[0x0001bc04]> s 0x0001bc98
[0x0001bc98]> ab
jump: 0x0001bcf8
opaddr: 0x0001bc98
addr: 0x0001bc98
size: 4
inputs: 2
outputs: 1
ninstr: 2
traced: false
[0x0001bc98]> pd 3
│           0x0001bc98      d00763e8       ld [i5+0x3e8], o0           ; arg6
            0x0001bc9c      80a68008       cmp i2, o0
        ┌─< 0x0001bca0      02800016       be 0x1bcf8                  ; sym.error_at_line+0xf4

```

The file is part of the testbins, found [here](https://github.com/radareorg/radare2-testbins/blob/master/elf/elf-solaris-sparc-ls)

### Additional Logs, screenshots, source-code,  configuration dump, ...

I don't know sparc. I found this while trying to write a regression test. I am reporting because it is a bug, not because I personally need it fixed.

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Test plan**

Tests pass

**Closing issues**

Closes #17298

### CODE REVIEW

**Comment by GustavoLCR** [comment]
Wtf github ci getting wrong commit

File: libr/anal/fcn.c [hunk 1]
```diff
@@ -336,6 +336,18 @@ static RAnalBlock *bbget(RAnal *anal, ut64 addr, bool jumpmid) {
 		if (((bb->addr >= eaddr && addr == bb->addr)
 		     || r_anal_block_contains (bb, addr))
 		    && (!jumpmid || r_anal_block_op_starts_at (bb, addr))) {
+			if (anal->opt.delay && bb->ninstr >= 3 && addr == (bb->addr + r_anal_bb_offset_inst (bb, bb->ninstr - 2))) {
+				RAnalOp op;
+				ut8 buf[32];
+				ut64 at = bb->addr + r_anal_bb_offset_inst (bb, bb->ninstr - 3);
+				anal->iob.read_at (anal->iob.io, at, buf, sizeof (buf));
+				int size = r_anal_op (anal, &op, at, buf, sizeof (buf), R_ANAL_OP_MASK_BASIC);
+				if (size && op.delay) {
```
Comment (trufae) [inline libr/anal/fcn.c:345]: op.delay is a counter, must loop X times

Comment (trufae) [reply libr/anal/fcn.c:345]: check size > 0

**Comment by XVilka** [comment]
"Analize" has a slightly different meaning.

**Pull request merged**